src/core/api.js

```diff
@@ -0,0 +1,17 @@
+import { http } from "./http";
+
+export function fetchPages() {
+    return http.get("/uiapi/page");
+}
+
+export function createPage(data) {
+    return http.post("/uiapi/page", data);
+}
+
+export function updatePage({ pageId, data }) {
+    return http.patch(`/uiapi/page?id=${pageId}`, data);
```

Comment:
@rvijendran-sailthru FYI the frontend is using PATCH for now. This is changeable.

Reply:
This is subject to change based on decisions for the backend.

Reply:
there may also be a way to pass params via function args

Reply:
there is but it is the 4th function arg

```suggestion
    return http.patch(`/uiapi/page`, data, , { id: pageId } );
```

Reply:
(not sure if javascript supports empty args like in that example)